discovery/seata-discovery-sofa/src/main/java/io/seata/discovery/registry/sofa/SofaRegistryServiceImpl.java

```diff
@@ -131,12 +133,13 @@ private RegistryClient getRegistryInstance() {
                         .setRegistryEndpoint(StringUtils.substringBefore(address, HOST_SEPERATOR))
                         .setRegistryEndpointPort(Integer.parseInt(portStr)).build();
 
-                    registryClient = new DefaultRegistryClient(config);
-                    ((DefaultRegistryClient)registryClient).init();
+                    result = new DefaultRegistryClient(config);
+                    ((DefaultRegistryClient) result).init();
+                    registryClient = result;
```

Comment:
You can continue to return registryClient, and result is used as a local transit variable, keeping the minimum changes.

Reply:
@zjinlei 
This way minimizes the number of volatile field reads, which improves performance, but makes the code a bit complicated. 

If minimum change or simple code is preferred, I will change the code to what you recommend ～

Reply:
[RegistryClient result = registryClient] == [null == registryClient]
[result = registryClient] == [null == registryClient]
Same reading times, and here is singleton initialization, code readability is more important.

Reply:
@zjinlei 
Just to confirm. Do you mean to change it like this?
```java
if (null == registryClient) {
    synchronized (SofaRegistryServiceImpl.class) {
        if (null == registryClient) {
            // ...
            DefaultRegistryClient result = new DefaultRegistryClient(config);
            result.init();
            registryClient = result;
        }
    }
}
return registryClient;
```

The original code;
```java
RegistryClient result = registryClient;
if (null == result) {
    synchronized (SofaRegistryServiceImpl.class) {
        result = registryClient;
        if (null == result) {
            // ...    
            result = new DefaultRegistryClient(config);
            ((DefaultRegistryClient) result).init();
            registryClient = result;
        }
    }
}
return result;
```